src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectUserId, setselectUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [listTodos, setListTodos] = useState<Todo[]>(todos);
+
+  const [hasNameError, setHasNameError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const valideData = `zxcvbnmasdfghjklqwertyuiopZXCVBNMASDFGHJKLQWERTY
```

Comment:
will be better to handle it via some regular expression instead of creating string with allowed symbols manually
https://stackoverflow.com/questions/388996/regex-for-javascript-to-allow-only-alphanumeric